src/pocketmine/entity/object/FallingBlock.php

```diff
@@ -70,6 +81,10 @@ protected function initEntity(CompoundTag $nbt) : void{
 		$damage = $nbt->getByte("Data", 0);
 
 		$this->block = BlockFactory::get($blockId, $damage);
+		if ($this->block instanceof Anvil) {
+			$this->behaviours = [new DestroyWhileFalling($this, [Block::TORCH, Block::COLORED_TORCH_RG, Block::COLORED_TORCH_BP,
+					Block::LIT_REDSTONE_TORCH, Block::UNLIT_REDSTONE_TORCH, Block::UNDERWATER_TORCH], true)];
```

Comment:
I don't like this. I'm trying to get rid of legacy block IDs. In addition, you're creating a whole new system here for one thing, which makes no sense and is inconsistent with the rest of the code.

It seems like this logic can be dropped and just add `canBlockLand()` implementations for these blocks.

Reply:
In fact, the logic for this already exists, so this code can be removed entirely.